src/visualization/utils/formatStatValue.ts

```diff
@@ -35,19 +35,14 @@ export const formatStatValue = (
 
   digits = Math.min(digits, MAX_DECIMAL_PLACES)
 
-  if (typeof value === 'number') {
-    const [wholeNumber, fractionalNumber] = Number(value)
-      .toFixed(digits)
-      .split('.')
-
-    localeFormattedValue = Number(wholeNumber).toLocaleString(undefined, {
-      maximumFractionDigits: MAX_DECIMAL_PLACES,
-    })
-
-    if (fractionalNumber) {
-      localeFormattedValue += `.${fractionalNumber}`
-    }
-  } else if (typeof value === 'string') {
+  const formatter = Intl.NumberFormat(undefined, {
+    minimumFractionDigits: digits,
+    maximumFractionDigits: digits,
+  })
+
+  if (isNumber(value)) {
+    localeFormattedValue = formatter.format(Number(value))
+  } else if (isString(value)) {
     localeFormattedValue = value
   } else {
```

Comment:
This is not needed to fix the issue, however, we are adding this change for posterity. We want this change in the history before the whole file is removed (in a separate pull request).

Reply:
Also, there is a minor difference. Giraffe uses utils `isNumber` and `isString` while here in the UI we simply use `typeof` to check the type.

Reply:
>We want this change in the history before the whole file is removed (in a separate pull request).

why do we want that?